test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -198,6 +202,32 @@ func (r *DockerMachineReconciler) reconcileNormal(machine *clusterv1.Machine, do
 	providerID := externalMachine.ProviderID()
 	dockerMachine.Spec.ProviderID = &providerID
 
+	// Set the nodeRef on the machine
+	nodeList := &v1.NodeList{}
+	if err := r.Client.List(ctx, nodeList); err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to list noderef")
+	}
+	pid, err := noderefutil.NewProviderID(providerID)
+	if err != nil {
+		r.Log.Error(err, "failed to parse node's provider spec")
+		return ctrl.Result{}, err
+	}
+	for _, node := range nodeList.Items {
+		npid, err := noderefutil.NewProviderID(node.Spec.ProviderID)
+		if err != nil {
+			r.Log.Error(err, "failed to parse node's provider spec")
+			continue
+		}
+		if pid.Equals(npid) {
+			machine.Status.NodeRef = &v1.ObjectReference{
```

Comment:
This is done by CAPI - why do you need it in here too?

Reply:
oh right. god. i made this mistake before -_-

ok what actually happened is i forgot to undo this commit and then saw it in a larger PR and applied no thought and just opened this one. My fault.